Lib/functools.py

```diff
@@ -868,3 +868,50 @@ def _method(*args, **kwargs):
     @property
     def __isabstractmethod__(self):
         return getattr(self.func, '__isabstractmethod__', False)
+
+
+################################################################################
+### cached_property() - computed once per instance, cached as attribute
+################################################################################
+
+_NOT_FOUND = object()
+
+
+class cached_property:
+    def __init__(self, func):
+        self.func = func
+        self.attrname = None
+        self.__doc__ = func.__doc__
+        self.lock = RLock()
+
+    def __set_name__(self, owner, name):
+        if self.attrname is None:
+            self.attrname = name
+        else:
+            raise TypeError(
+                "Cannot assign the same cached_property to two names "
+                f"({self.attrname!r} and {name!r}) on the same class."
```

Comment:
This message is inaccurate because the exception is also raised when it's assigned on the different classes.
I'm not sure that we should disallow to assign it on the different classes under the same name.

Reply:
Good catch. I'd like to avoid tracking owners, but it should be sufficient to just allow repeated `__set_name__` as long as the name remains the same.